### PR TITLE
Automated releases with Travis

### PR DESCRIPTION
**Pull request by megastep**
This will make the task of pushing out releases a lot easier. Basically all you have to do is push out a new tag, and it will compile all the archives and upload them to the GitHub release tag, as well as to the S3 bucket for CocoaPods.

The GitHub secret maps to my account so it should be able to push releases to the main AudioKit repo. I am setting up the Travis environment variables with the authentication for the S3 bucket too.

Basically all you'll have to do to push out a new release will be something like this:

`git tag v2.4.4`
`git push --tags`
 ... and once that's done building, you can just update the release notes and push the pod!

### CODE REVIEW

**Comment by megastep** [comment]
Also: a full release build takes about 30 minutes on Travis.

**Comment by aure** [comment]
Looks fantastic, looking forward to trying it out. Perhaps tonight?
